lib/src/mui_suggestors/components/chip/badge_to_mui_chip_migrator.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:over_react_codemod/src/mui_suggestors/components/chip/shared.dart';
+import 'package:over_react_codemod/src/mui_suggestors/components/shared_props_migrators/color_props.dart';
+import 'package:over_react_codemod/src/mui_suggestors/components/shared_props_migrators/hit_area.dart';
+import 'package:over_react_codemod/src/mui_suggestors/constants.dart';
+import 'package:over_react_codemod/src/util/component_usage.dart';
+import 'package:over_react_codemod/src/util/component_usage_migrator.dart';
+
+import '../mui_migrator.dart';
+
+class BadgeToMuiChipMigrator extends ComponentUsageMigrator
+    with
+        MuiMigrator,
+        BadgeToMuiChipPropsMigrator,
+        MuiChipPropsMigrator,
+        HitAreaPropMigrators,
+        ColorPropMigrators {
+  @override
+  bool shouldMigrateUsage(FluentComponentUsage usage) =>
+      usesWsdFactory(usage, 'Badge');
+
+  @override
+  void migrateUsage(FluentComponentUsage usage) {
+    super.migrateUsage(usage);
+
+    yieldPatchOverNode('$muiNs.Chip', usage.factory!);
+    migrateChildrenToLabelProp(usage);
+
+    handleCascadedPropsByName(usage, {
+      'align': migrateBadgeAlignProp,
+      'backgroundColor': (p) => migrateColorPropsBackgroundColor(usage, p),
+      'borderColor': yieldUseSxFixmePatch,
+      'isOutline': (p) => migrateBadgeOutlineProp(usage, p),
+      'textColor': yieldUseSxFixmePatch,
+
+      // HitArea Props
+      'allowedHandlersWhenDisabled': yieldPropManualMigratePatch,
+      'isDisabled': (p) {
+        yieldPropFixmePatch(p,
+            'if this component has mouse handlers that should fire when disabled or needs to show a tooltip/overlay when disabled, add a wrapper element');
+        yieldPropPatch(p, newName: 'disabled');
+      },
+      'role': (p) => yieldPropPatch(p, newName: 'dom.role'),
+      'target': (p) => yieldPropPatch(p, newName: 'dom.target'),
```

Comment:
Wouldn't `href` need something similar? Or is `href` a prop on a MUI `Chip`?

Reply:
Ope yep, we do need a similar handling for `href`. Good catch!